src/input.cpp

```diff
@@ -1187,6 +1187,17 @@ std::string input_context::describe_key_and_name( const std::string &action_desc
     return get_desc( action_descriptor, get_action_name( action_descriptor ), evt_filter );
 }
 
+// In ImGui, there are lots of cases where we need to execute code related to an action while drawing.
+//  We can't always do this cleanly during drawing, so we set action_override here so that the next
+//  time input_context::handle_input gets no input, it instead returns action_override.
+static std::string action_override;
```

Comment:
Probably need to skip `inp_mngr.get_timeout` altogether when there is an action override so the next frame is drawn immediately without waiting for further input, to reflect any changes caused by the overridden action. Previously I saw you added zero timeout to some inventory UI input contexts, was that for input responsiveness? If so this might be the cause.

Reply:
action_override got nixed in recent commits, I found a nicer way of doing what I was trying to do without modifying input_context